src/XComponent.Functions.Test/FunctionsManagerTest.cs

```diff
@@ -168,14 +168,51 @@ public interface ISender
                 PublicMember = publicMemberAfter,
             };
 
+            var sender = new object(); 
+            var context = new object();
             var functionsManager = new FunctionsManager("component", "statemachine");
+            functionsManager.AddTask(null, null, null, context, sender);
             functionsManager.ApplyFunctionResult(functionResult,
                     publicMemberBefore,
                     null,
-                    new object(),
-                    new object());
+                    context,
+                    sender);
 
             Assert.AreEqual("after", publicMemberBefore.State);
         }
+
+        [Test]
+        public void ApplyFunctionResultShouldThrowSerializationException()
+        {
+            var publicMemberBefore = new PublicMember() { State = "before" };
+
+            var publicMemberAfter = new JObject();
+            publicMemberAfter.Add("State", new JValue("after"));
+
+            var functionResult = new FunctionResult()
+            {
+                PublicMember = "Hello",
+            };
+
+            var functionsManager = new FunctionsManager("component", "statemachine");
+            
+            Assert.Throws<SerializationException>(() => functionsManager.ApplyFunctionResult(functionResult,
+                publicMemberBefore,
+                null,
+                new object(),
+                new object()));
+        }
+
+        [Test]
+        public void ApplyFunctionResultShouldThrowSenderNotFoundException()
+        {
+            var functionsManager = new FunctionsManager("component", "statemachine");
+
+            Assert.Throws<Exception>(() => functionsManager.ApplyFunctionResult(null,
```

Comment:
test the message in the exception.

like if it contains the words 'not found' and 'sender'

Reply:
I don't get this test... why is the sender not found? there's no sender in your result!!
as I said before, we need two tests:

1) there's no sender in the result, and we should test if **no exception is thrown**
2) there's a sender in the result whose name doesn't match the ones in the sender object, and we should test if **an exception is thrown**

In the error message please put **exactly** the signature of the method it is looking for (as as much information as possible).